### PR TITLE
Extract CSS from JS

### PR DESCRIPTION
**Pull request by BeatC**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -124,5 +127,47 @@ module.exports = [
         output: Object.assign({}, config.output, {
             library: 'JitsiMeetExternalAPI'
         })
-    })
+    }),
+
+    // React application
```
Comment (lyubomir) [inline webpack.config.js]: We don't want a new entry point (respectively, Webpack configuration). We want to modify the implementation of the existing entry point `app.bundle` without changing its export i.e. it should remain `APP` and the jitsi-meet application should continue to work (as seen from the outside) after merging your PR without loss of functionality, as if we haven't introduced React. I've merged the HasteResolverPlugin and the babel presets into the existing entry point `app.bundle` so you'll have to make sure that (a minimal set of) the remaining modifications are incorporated there as well.

File: css/_filmstrip.scss [hunk 2]
```diff
@@ -60,6 +60,7 @@
         font-size: 0pt; /*!!!Removes the gap between the local video container and the remote videos.*/
 
         &.hidden {
+            display: block; // TOFIX: overwrites aui styles
```
Comment (lyubomir) [inline css/_filmstrip.scss]: Could you please give me a bit more context about this change? Is it necessary because of React work being done in this PR? Or is it a separate fix unrelated to the React effort that you happened to discover while working?

File: index.html [hunk 3]
```diff
@@ -29,12 +29,12 @@
                 window.onload = function() {
                     document.body.innerHTML
                         = "The application failed to load, missing file: "
-                            + fileRef;
+                        + fileRef;
```
Comment (lyubomir) [inline index.html]: I disagree with the indentation change introduced here, please remove it from this PR.

File: package.json [hunk 4]
```diff
@@ -49,23 +54,26 @@
...
     "clean-css": "*",
-    "css-loader": "*",
+    "css-loader": "^0.26.0",
     "eslint": ">=3",
     "eslint-plugin-jsdoc": "*",
     "eslint-plugin-react": "*",
     "eslint-plugin-react-native": "*",
     "expose-loader": "*",
-    "file-loader": "*",
+    "file-loader": "^0.9.0",
```
Comment (lyubomir) [inline package.json]: I need to know how important it is to this PR that we mention versions. I know that there are generally pros to mentioning minimal versions but I'm not sure that I want to address these (possibly) best practices in the scope of this PR. If you feel strongly about them but they are not essential to this PR, let's please have a separate PR for them.

File: react/features/conference/components/Conference.web.js [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+const displayNoneStyle = {
+    display: 'none'
+};
+
+/**
+ * Implements a React Component which renders initial conference layout
+ *
+ * @returns { ReactElement }
+ */
+const Conference = () =>
```
Comment (lyubomir) [inline react/features/conference/components/Conference.web.js]: Could I please have the React components defined as `class` like the previous/existing ones rather than as plain functions? I understand that they look simple as plain functions at the moment but (1) they are different from the previous/existing ones thus creating inconsistency, (2) more importantly, we'll have a harder time later on when/if we decide to add more methods to them, and (3) we want propTypes on them.

File: index.html [hunk 6]
```diff
@@ -44,152 +44,19 @@
...
-    </div>
+    <div id="root"></div>
     <div id="keyboard-shortcuts" class="keyboard-shortcuts" style="display:none;">
         <div class="content">
             <ul id="keyboard-shortcuts-list" class="shortcuts-list">
             </ul>
         </div>
     </div>
+    <script src="libs/main.min.js"></script>
+    <script src="libs/app.bundle.min.js?v=139"></script>
```
Comment (lyubomir) [inline index.html]: I'll think about the split between main.min.js and app.bundle.min.js tomorrow because I'd like us to try to avoid it.

**Comment by BeatC** [comment]
jenkins test this please

File: css/_filmstrip.scss (hunk 2 above)
Comment (BeatC) [inline css/_filmstrip.scss]: This line is fixing bug introduced after I move `bundle.js` to body. As we link AUI css from `bundle.js` it has more specificity than our CSS file (`all.css`). There are more bugs introduced by this change. I think we can fix it by linking to AUI styles from `index.html` instead of `bundle.js`

File: react/features/conference/components/Conference.web.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+ */
+export default class Conference extends Component {
+
+   /**
+    * Implements React's {@link Component#render()}.
+    *
+    * @inheritdoc
+    * @returns {ReactElement}
+    */
+    render() {
```
Comment (lyubomir) [inline react/features/conference/components/Conference.web.js]: In at least 4 of the .web.js files I notice that the indentation of method documentation comments is incorrect/inconsistent. In the future, please make sure that the source code of your PRs including its formatting is consistent with the existing source code.

Please don't modify this PR until further notice because I'm currently trying to merge parts of it!

File: .editorconfig [hunk 8]
```diff
@@ -8,6 +8,3 @@ indent_size = 4
 indent_style = space
 max_line_length = 80
 trim_trailing_whitespace = true
-
-[Makefile]
-indent_style = tab
```
Comment (lyubomir) [inline .editorconfig]: Why do you want the above removed? You introduced it in this PR, I extracted it into a separate commit because I didn't want to pollute this PR with unrelated changes, and I merged it into master.

File: Makefile [hunk 9]
```diff
@@ -2,6 +2,7 @@ BUILD_DIR = build
 CLEANCSS = ./node_modules/.bin/cleancss
 DEPLOY_DIR = libs
 LIBJITSIMEET_DIR = node_modules/lib-jitsi-meet/
+AUI_DIR = node_modules/@atlassian/aui
```
Comment (lyubomir) [inline Makefile]: Why do you want the above added? You don't seem to be using the variable.

File: webpack.config.js [hunk 10]
```diff
@@ -2,19 +2,22 @@
 
 require('babel-polyfill'); // Define Object.assign() from ES6 in ES5.
 
+var webpack = require('webpack');
 var HasteResolverPlugin = require('haste-resolver-webpack-plugin');
+var ExtractTextPlugin = require('extract-text-webpack-plugin');
 var process = require('process');
-var webpack = require('webpack');
```
Comment (lyubomir) [inline webpack.config.js]: Why do you want the definition of the variable webpack moved?

File: package.json [hunk 11]
```diff
@@ -21,13 +21,14 @@
...
     "events": "*",
+    "extract-text-webpack-plugin": "^1.0.1",
     "haste-resolver-webpack-plugin": "^0.2.2",
     "i18next": "3.4.4",
     "i18next-xhr-backend": "1.1.0",
+    "jQuery-Impromptu": "trentrichardson/jQuery-Impromptu#v6.0.0",
     "jquery": "~2.1.1",
     "jquery-contextmenu": "*",
     "jquery-i18next": "1.1.0",
-    "jQuery-Impromptu": "trentrichardson/jQuery-Impromptu#v6.0.0",
```
Comment (lyubomir) [inline package.json]: Could I please ask you to remove the move of jQuery-Impromptu from this PR?

File: .editorconfig (hunk 8 above)
Comment (BeatC) [inline .editorconfig]: I misunderstood you. Ok, I'll create another PR with this changes.

Comment (BeatC) [inline .editorconfig]: OK, I misunderstood you. Great, it would be useful.

File: webpack.config.js (hunk 10 above)
Comment (BeatC) [inline webpack.config.js]: Maybe it was moved after rebase. You and me introduced this changes in the file, so I removed one of the lines.

File: package.json (hunk 11 above)
Comment (BeatC) [inline package.json]: Yes, sure. This probably was caused because I use Yarn instead of npm (yarn is definitely faster) and it could change the order of dependencies in `package.json`

File: Makefile (hunk 9 above)
Comment (BeatC) [inline Makefile]: I forgot to remove this variable. I used it for bundling aui css's

File: package.json (hunk 11 above)
Comment (BeatC) [inline package.json]: Yes, done

File: index.html (hunk 3 above)
Comment (BeatC) [inline index.html]: Removed this change

File: package.json (hunk 4 above)
Comment (BeatC) [inline package.json]: Updated versions in `package.json`

File: react/features/base/util/loadScript.web.js [hunk 12]
```diff
@@ -0,0 +1,84 @@
+/**
```
Comment (BeatC) [inline react/features/base/util/loadScript.web.js]: I propose to merge this file when we move conference initialization to react app

Comment (lyubomir) [inline react/features/base/util/loadScript.web.js]: OK. Do you mean that you'll remove it from this PR? I don't think it's necessary here but I'm merely asking to better understand your proposal.

The biggest problem that I have this file/function though is that there's one like it in lib-jitsi-meet. I feel like we should reuse it and have 1 in the 2 projects. What do you think?
